lib/driver.js

```diff
@@ -233,6 +238,36 @@ class EspressoDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Unzip the given app path and return the first package that has SUPPORTED_EXTENSIONS
+   * in the archived file.
+   *
+   * @param {string} appPath The path to app file.
+   * @returns {string} Retuns the path to an unzipped app file path.
+   * @throws Raise an exception if the zip did not have any SUPPORTED_EXTENSIONS packages.
+   */
+  async unzipApp (appPath) {
+    const useSystemUnzipEnv = process.env.APPIUM_PREFER_SYSTEM_UNZIP;
+    const useSystemUnzip = _.isEmpty(useSystemUnzipEnv)
+      || !['0', 'false'].includes(_.toLower(useSystemUnzipEnv));
+    const tmpRoot = await tempDir.openDir();
+    await zip.extractAllTo(appPath, tmpRoot, {useSystemUnzip});
+
+    const globPattern = `**/*.+(${SUPPORTED_EXTENSIONS.map((ext) => ext.replace(/^\./, '')).join('|')})`;
+    const sortedBundleItems = (await fs.glob(globPattern, {
+      cwd: tmpRoot,
+      strict: false,
+    })).sort((a, b) => a.split(path.sep).length - b.split(path.sep).length);
+    if (sortedBundleItems.length === 0) {
+      // no expected packages in the zip
+      logger.errorAndThrow(`${this.opts.app} did not have any of '${SUPPORTED_EXTENSIONS.join(', ')}' ` +
+      `extension packages. Please make sure the zip file has a valid package.`);
```

Comment:
Please make sure the provided .zip archive contains at least one valid package bundle